tracks/ruby/exercises/matrix/README.md

```diff
@@ -0,0 +1,35 @@
+# Matrix
+
+## Reasonable Solutions
+
+```ruby
+class Matrix
+  attr_reader :rows, :columns
+  def initialize(input)
+    @rows = split_input(input)
+    @columns = rows.transpose
+  end
+
+  private
+
+  def split_input(string)
+    string.lines.map { |line| split_line(line) }
+  end
+
+  def split_line(line)
+    line.split.map(&:to_i)
+  end
+end
+```
+
+Many solutions will not split out the input parsing into methods like this. We don't yet have consensus among mentors if this is important or not. If the methods are split, the above example does demonstrate how the methods could be flagged as private. Some Ruby projects insist on marking as private methods that are not part of the public, tested interface.
+
+## Common Suggestions
+
+There's no need to overload a student with all of these suggestions, so here is a list ordered roughly by most important suggestions first:
+
+* For students who wrote code that interleaves string manipulation with data persistance, encourage them to do those steps sequentially
+* Suggest the student look at other methods on `String` that could simplify their code if they don't use `String#lines`
+* More generally, for any solutions that has matching based on a regular expression or explicitly decleared string or substring, encourage them to find a solution that does not require either
+* For solutions that do not iterate over the output of `String#lines`, suggestions common to the Hamming exercise may also work well
+* For solutions that use `Array#transpose`, you can challenge the student to solve the problem without it, and vice versa
```

Comment:
I feel very strongly about this. Exercism is about students practicing to getting fluent in a language, not about general programming skills. Especially at this point in the core exercises progression, the goal for students is to find out that there are so many convenience methods in Ruby, and _that_ makes it a breeze to work with. For me, asking students to implement functions that Ruby provides is out of scope. 
For deeper insight, it would suffice to point them to the source code, and let them read how it's implemented.

Reply:
I understand the general point about practicing Ruby fluency rather than algorithmic problem-solving as such. However, coming up with a nice Rubyish implementation of `transpose` (pretending for a second that "it wasn't there") could also be helpful to build Ruby fluency, couldn't it? In fact, isn't reimplementing part of a language's standard library a pretty standard learner's approach to gaining fluency in that language?

Reply:
> coming up with a nice Rubyish implementation of transpose (pretending for a second that "it wasn't there") could also be helpful to build Ruby fluency, couldn't it? 

That's an interesting question and I don't have a final answer :-)  I agree that implementing some standard language features can be interesting and yes, one can gain insight from it. One such thing could be to reimplement the `inject`, `select` and other Power Enumerables with `each`. I'm not convinced that it's within the scope of Exercism, but I can see the fun of having some side exercises with such problems. 

> isn't reimplementing part of a language's standard library a pretty standard learner's approach to gaining fluency in that language?

I sincerely doubt that it's the most _efficient_ way to gain fluency, but let's go with that it is a valid approach, then I still think that it should come later in the track, when student's have gained 'daily fluency'. 
 
My pledge here is: that it should not be part of a core exercise, just because that exercise uses a convenience method.  And that _if_ we want such an reimplementing exercise, the implementing should be an interesting exercise in itself.  

Thoughts?